### YamlMime:ManagedReference
items:
- uid: System.Threading.ThreadStateException
  id: ThreadStateException
  children:
  - System.Threading.ThreadStateException.#ctor
  - System.Threading.ThreadStateException.#ctor(System.String)
  - System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: ThreadStateException
  nameWithType: ThreadStateException
  fullName: System.Threading.ThreadStateException
  type: Class
  summary: "ときにスローされる例外を<xref href=&quot;System.Threading.Thread&quot;></xref>が無効です<xref:System.Threading.Thread.ThreadState*>メソッド呼び出しの</xref:System.Threading.Thread.ThreadState*>。"
  remarks: "少なくとも&1; つには、スレッドが作成されたら、<xref:System.Threading.ThreadState>に終了するまでの状態します</xref:System.Threading.ThreadState>。 `ThreadStateException`できません、スレッドの現在の状態のため、要求された操作を実行するメソッドによってスローされます。 たとえばを呼び出すことによって中止されたスレッドを再起動しようとしている<xref:System.Threading.Thread.Start%2A>が終了したスレッドでスロー、ThreadStateException</xref:System.Threading.Thread.Start%2A> 。       ThreadStateException は、値は 0x80131520 HRESULT COR_E_THREADSTATE を使用します。       ThreadStateException のインスタンスの初期プロパティ値の一覧は、次を参照してください、<xref:System.Threading.ThreadStateException.%23ctor%2A>コンス トラクター。</xref:System.Threading.ThreadStateException.%23ctor%2A> 。"
  example:
  - "The following example demonstrates an error that causes the system to throw a `ThreadStateException`.  \n  \n [!code-cpp[ThreadStEx#1](~/add/codesnippet/cpp/t-system.threading.threa_3_1.cpp)]\n [!code-vb[ThreadStEx#1](~/add/codesnippet/visualbasic/t-system.threading.threa_3_1.vb)]\n [!code-cs[ThreadStEx#1](~/add/codesnippet/csharp/t-system.threading.threa_3_1.cs)]  \n  \n This code produces the following output:  \n  \n```  \nIn main. Attempting to restart myThread.  \n    Working thread...  \n    Caught: Thread is running or terminated. Cannot restart.  \n```"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ThreadStateException : SystemException
  inheritance:
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor
  id: '#ctor'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException()
  nameWithType: ThreadStateException.ThreadStateException()
  fullName: System.Threading.ThreadStateException.ThreadStateException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "新しいインスタンスを初期化、 <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref>既定のプロパティを持つクラス。"
  remarks: "このコンス トラクター、 `Message` 「要求された操作は、現在の状態により、スレッドで実行できません」. などに、エラーを説明するシステム提供のメッセージの新しいインスタンスのプロパティ このメッセージでは、現在のシステム カルチャが考慮されます。       次の表は、 <xref:System.Threading.ThreadStateException>。</xref:System.Threading.ThreadStateException>のインスタンスの初期プロパティ値を示しています。      |プロパティ |値 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|`null`.|  |<xref:System.Exception.Message%2A>|ローカライズされたエラー メッセージ文字列 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A> 。"
  syntax:
    content: public ThreadStateException ();
    parameters: []
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(String)
  nameWithType: ThreadStateException.ThreadStateException(String)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "新しいインスタンスを初期化、 <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref>指定したエラー メッセージを使用します。"
  remarks: "このコンス トラクター、`Message`の新しいインスタンスを使用して、プロパティ、`message`パラメーター。       次の表は、 <xref:System.Threading.ThreadStateException>。</xref:System.Threading.ThreadStateException>のインスタンスの初期プロパティ値を示しています。      |プロパティ |値 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|`null`.|  |<xref:System.Exception.Message%2A>|エラー メッセージ文字列 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A> 。"
  syntax:
    content: public ThreadStateException (string message);
    parameters:
    - id: message
      type: System.String
      description: "例外の原因を説明するエラー メッセージ。"
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(SerializationInfo,StreamingContext)
  nameWithType: ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "新しいインスタンスを初期化、 <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref>シリアル化されたデータを持つクラス。"
  remarks: "このコンス トラクターは、ストリームで送信された例外オブジェクトを再構成する逆シリアル化中に呼び出されます。 詳細については、次を参照してください。 [XML および SOAP シリアル化](~/add/includes/ajax-current-ext-md.md)です。"
  syntax:
    content: protected ThreadStateException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "<xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;> </xref>スローされた例外に関するシリアル化されたオブジェクト データを保持します。"
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "<xref:System.Runtime.Serialization.StreamingContext>ソースまたは転送先に関するコンテキスト情報を含む</xref:System.Runtime.Serialization.StreamingContext>。"
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(String,Exception)
  nameWithType: ThreadStateException.ThreadStateException(String,Exception)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "新しいインスタンスを初期化、 <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref>指定したエラー メッセージおよびこの例外の原因となった内部例外への参照を持つクラス。"
  remarks: "前の例外への参照を含める必要があります前の例外の直接の結果としてスローされる例外、<xref:System.Exception.InnerException%2A>プロパティ</xref:System.Exception.InnerException%2A>。 <xref:System.Exception.InnerException%2A>プロパティは、コンス トラクターに渡されたものと同じ値を返しますまたは`null`場合、<xref:System.Exception.InnerException%2A>プロパティがコンス トラクターには、内部例外値を指定していません</xref:System.Exception.InnerException%2A></xref:System.Exception.InnerException%2A>。       次の表は、 <xref:System.Threading.ThreadStateException>。</xref:System.Threading.ThreadStateException>のインスタンスの初期プロパティ値を示しています。      |プロパティ |値 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|内部例外の参照 |。  |<xref:System.Exception.Message%2A>|エラー メッセージ文字列 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A> 。"
  syntax:
    content: public ThreadStateException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "例外の原因を説明するエラー メッセージ。"
    - id: innerException
      type: System.Exception
      description: "現在の例外の原因となった例外。 場合、`innerException`パラメーターではありません<xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;></xref>で現在の例外が発生、 <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref>ブロックは、内部例外を処理します。"
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.Threading.ThreadStateException.#ctor
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException()
  nameWithType: ThreadStateException.ThreadStateException()
  fullName: System.Threading.ThreadStateException.ThreadStateException()
- uid: System.Threading.ThreadStateException.#ctor(System.String)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(String)
  nameWithType: ThreadStateException.ThreadStateException(String)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(SerializationInfo,StreamingContext)
  nameWithType: ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(String,Exception)
  nameWithType: ThreadStateException.ThreadStateException(String,Exception)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Threading.ThreadStateException.#ctor*
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException
  nameWithType: ThreadStateException.ThreadStateException
